mockgcp/mockbigqueryconnection/connection.go

```diff
@@ -201,18 +201,8 @@ func (s *ConnectionV1) UpdateConnection(ctx context.Context, req *pb.UpdateConne
 	if err := s.storage.Get(ctx, fqn, obj); err != nil {
 		return nil, err
 	}
-
-	paths := req.GetUpdateMask().GetPaths()
-	for _, path := range paths {
-		switch path {
-		case "friendlyName":
-			obj.FriendlyName = req.GetConnection().GetFriendlyName()
-		case "description":
-			obj.Description = req.GetConnection().GetDescription()
-		default:
-			return nil, status.Errorf(codes.InvalidArgument, "update_mask path %q not valid", path)
-		}
-	}
+	obj.FriendlyName = req.GetConnection().GetFriendlyName()
```

Comment:
Not sure I follow the change. In the new mock implementation we are always setting these 2 fields?

Is this behavior specific to the real API? or do we plan to improve the mock in the future?

Reply:
Are you asking why not checking the update field_mask? This is an easy hack because the BQCC controller now uses the `CompareProtoMessage`!

More context:
In general, if the GCP server supports update field mask,  the MockGCP needs to support each field mask as well to tell if a field can be changed or not. However, this requires a lot of engineering effort (boring, repeated code). I made some improvements recently that the controller can determine the field immutable in the `CompareProtoMessage` function via the proto, so that contributors no longer need to repeatedly write `reflect.DeepEqual` ... `if fieldPath[i] == "display_name" in either MockGCP and scifi controller. Corresponding code (https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/pkg/controller/direct/common/compare.go#L77C13-L77C28),

Reply:
The helper library looks very cool! Removing repeated code and effort sounds exciting. My only concern is that it might affect the mock's ability to guard the controller's behavior in presubmit tests?

Reply:
I think we only want to copy the values if the field path is specified in the update field_mask

Reply:
Updated. Also added a handler to only update the field based on field_mask